pkg/progressreader/progressstatus.go

```diff
@@ -0,0 +1,78 @@
+package progressreader
+
+import (
+	"bytes"
+	"io"
+	"sync"
+)
+
+// ProgressStatus keeps track of one or more observers watching the progress
+// of an operation. For example, if multiple clients are trying to pull an
+// image, they share a ProgressStatus for the download operation.
+type ProgressStatus struct {
```

Comment:
How about `Broadcaster`? Since it's in the `progressreader` namespace, `progressreader.Broadcaster` should make it pretty clear what it's for.